app/src/main/scala/org/alephium/explorer/api/BlockEndpoints.scala

```diff
@@ -22,33 +22,32 @@ import sttp.tapir._
 import sttp.tapir.generic.auto._
 
 import org.alephium.api.{alphJsonBody => jsonBody}
-import org.alephium.explorer.api.BaseEndpoint
 import org.alephium.explorer.api.model._
 import org.alephium.protocol.model.BlockHash
 
-trait BlockEndpoints extends BaseEndpoint with QueryParams {
+object BlockEndpoints extends BaseEndpoint with QueryParams {
 
-  private val blocksEndpoint =
+  private def blocksEndpoint() =
```

Comment:
if there are no differences, what do you think of defining:

```
private def blocksEndpoint
```
without the `()`? we wouldn't need to add them everywhere, as we wont' have args anyway.

This makes it a bit nicer IMO, but maybe I miss something?

Reply:
Just a preference. Parens `()` shows at the call site that what's being called is a function. If this funtion does need parameters in the future then changing it makes the call site a curried function which in some odd cases would skips compile check. 

It's just a preference thing. I use parens `()` in my projects mostly because it's self documenting. 

No biggie. Parens removed.

Reply:
`omission of parentheses` can be dangerous as the function might has side effects.

When `val` is possible, I believe that `(lazy) val` better than `def blocksEndpoint()` better than `def blocksEndpoint`. With `val` I am sure that it will be initialized only once, while `def` cannot.

The cost is just a few object references (4 bytes each). The extra cost is pretty low. I would consider removing those references as over-optimizations.

Reply:
It's as simple as this:
1. I provide my feedback.
2. If the feedback makes sense, then let's take it. If it does not make sense, just comment back and let me know.

It's never `whatever you want`.

Reply:
I think we both agree that `do whatever you want` and `do whatever I want` are both too extreme and bad for the project.

Reply:
The choice of words in your feedback is not starting a discussion. It’s telling to do something you think is right.

You have strong opinions and I really have no desire to counter your decisions even if it doesn't make sense. It's just wasted time.

This PR is not `do whatever I want`. Like every other PR an issue gets created to confirm its validity or discussed on Slack with Thomas and only then a PR gets submitted.

Please let me know how you’d like this PR progressed because I don’t know.

Reply:
It's never `whatever you want`. -> It's never `whatever I want`.

I copied that phrase from your sentence and forgot to replace `you` with `I`.

Reply:
Re: how to progress

I only looked at the discussion in this thread, I did not look at the other code of this PR. It's not me to decide how this PR should move forward as I don't review it at all.

> You have strong opinions and I really have no desire to counter your decisions even if it doesn't make sense. It's just wasted time.

I don't know how this kind of collaboration could work for the project for the long term, but let's discuss this off Github.